api/api.go

```diff
@@ -0,0 +1,267 @@
+	api.Router.HandleFunc("/instances/{instanceId}", api.getInstance).Methods("GET")
+	// Internal API
+	api.Router.HandleFunc("/instances/{instanceId}", api.updateInstance).Methods("PUT")
+	api.Router.HandleFunc("/instances/{instanceId}/events", api.addEvent).Methods("PUT")
+	api.Router.HandleFunc("/instances/{instanceId}/dimensions/{dimension_name}/options/{value}", api.addDimension).Methods("PUT")
+	api.Router.HandleFunc("/instances/{instanceId}/dimensions/{dimension_name}/nodeid/{value}", api.addNodeID).Methods("PUT")
+	api.Router.HandleFunc("/instances/{instanceId}/dimensions", api.getDimensions).Methods("GET")
+
+	return &api
+}
+
+func (api *ImportAPI) addJob(w http.ResponseWriter, r *http.Request) {
+	newJob, error := models.CreateJob(r.Body)
+	if error != nil {
+		log.Error(error, log.Data{})
+		http.Error(w, "Bad client request received", http.StatusBadRequest)
+		return
+	}
+	if validationError := newJob.Validate(); validationError != nil {
+		log.Error(validationError, log.Data{})
+		http.Error(w, "Bad client request received", http.StatusBadRequest)
+		return
+	}
+	jobInstance, dataStoreError := api.dataStore.AddJob(api.host, newJob)
+	if dataStoreError != nil {
+		log.Error(dataStoreError, log.Data{"newJob": newJob})
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+	bytes, error := json.Marshal(jobInstance)
+	if error != nil {
+		log.Error(error, log.Data{"Job": newJob})
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+	setJSONContentType(w)
+	w.WriteHeader(http.StatusCreated)
+	_, writeError := w.Write(bytes)
+	if writeError != nil {
+		log.Error(writeError, log.Data{"jobInstance": jobInstance})
+		http.Error(w, internalError, http.StatusInternalServerError)
+		return
+	}
+	log.Info("created new import job", log.Data{"job": jobInstance})
+}
+
+func (api *ImportAPI) addUploadedFile(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	jobID := vars["jobId"]
+	uploadedFile, error := models.CreateUploadedFile(r.Body)
+	if error != nil {
+		log.Error(error, log.Data{"file": uploadedFile, "job_id": jobID})
+		http.Error(w, "bad client request received", http.StatusBadRequest)
+		return
+	}
+	dataStoreError := api.dataStore.AddUploadedFile(jobID, uploadedFile)
+	if dataStoreError != nil {
+		log.Error(dataStoreError, log.Data{"file": uploadedFile, "job_id": jobID})
+		setErrorCode(w, dataStoreError)
+		return
+	}
+	log.Info("added uploaded file to job", log.Data{"job_id": jobID, "file": uploadedFile})
+}
+
+func (api *ImportAPI) updateJob(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	jobID := vars["jobId"]
+	job, err := models.CreateJob(r.Body)
+	if err != nil {
+		log.Error(err, log.Data{"jobState": job, "job_id": jobID})
+		http.Error(w, "bad client request received", http.StatusBadRequest)
+	}
+	dataStoreError := api.dataStore.UpdateJobState(jobID, job)
+	if dataStoreError != nil {
```

Comment:
re use err, rather than creating a new variable type.

You can also one line this so:

```go
if err := api.dataStore.UpdateJobState(jobID, job); err != nil {
                log.Error(err, log.Data{"job": job, "job_id": jobID})
 		setErrorCode(w, err)
 		return
}